clarity/src/vm/database/structures.rs

```diff
@@ -474,6 +474,39 @@ impl<'db, 'conn> STXBalanceSnapshot<'db, 'conn> {
         };
     }
 
+    /// Lock `amount_to_lock` tokens on this account until `unlock_burn_height`.
+    /// After calling, this method will set the balance to a "LockedPoxTwo" balance,
+    ///  because this method is only invoked as a result of PoX2 interactions
```

Comment:
This doesn't appear to be happening in that `match self.balance` block?  Looks like you're merely altering an existing `LockedPoxOne` instead of converting it to a `LockedPoxTwo`.

Reply:
Yeah, that's a copy-pasta rustdoc, will update.